.changelog/3859.internal.md

```diff
@@ -0,0 +1 @@
+go: Bump Badger to v3
```

Comment:
Maybe a note about the (automatic) migration would be appropriate.

Reply:
Done, please take a look.

Reply:
Looks good, thanks!